utils/events.ts

```diff
@@ -0,0 +1,103 @@
+import { isBefore, isAfter } from 'date-fns'
+import { toDate } from 'date-fns-tz'
+import { formatUTC, weeksBetween } from './date'
+import { ApiEvent } from 'apiClient'
+
+export type WeeklyData = {
+  week: number
+  date: Date
+  prior: Date
+  events: ApiEvent[]
+}
+
+export const withinBounds = (start: Date, end: Date) => (e: ApiEvent) => {
+  const time = toDate(e.occurred_at)
+  return isAfter(time, start) && isBefore(time, end)
+}
+export const eventsBetween = (
+  start: Date,
+  end: Date,
+  events: ApiEvent[]
+): ApiEvent[] => events.filter(withinBounds(start, end))
+
+export function uniqueById(x: readonly ApiEvent[]) {
+  return x.reduce((agg: ApiEvent[], i: ApiEvent) => {
+    if (!agg.map(({ id }) => id).includes(i.id)) {
+      return agg.concat(i)
+    }
+    return agg
+  }, [])
+}
+
+export const getAllVisibleEvents = (weeklyData: WeeklyData[]) =>
+  weeklyData.reduce((agg: ApiEvent[], { events }) => {
+    return agg.concat(events)
+  }, [])
```

Comment:
`flatMap` here might be a bit cleaner.

Reply:
Good call, `flatMap` falls into the category of addendums-to-JS-which-I-recognize-but-rarely-recall